File: ios-sdk/Sources/ConsentSDK/Models/ConsentState.swift
import Foundation

/// Holds the current state of the consent popup flow.
public struct ConsentState {
    public var age: String = ""
    public var selectedPurposes: Set<String> = []
    public var purposes: [Purpose] = []
    public var language: String = "en"
    public var isLoading: Bool = false
    public var isSubmitting: Bool = false
    public var error: String? = nil
    public var isPopupVisible: Bool = false
    public var isConsentSubmitted: Bool = false
    public var validationErrors: [String: String] = [:]
}

/// Error codes for consent operations.
public enum ConsentErrorCode: String {
    case networkError = "NETWORK_ERROR"
    case timeout = "TIMEOUT"
    case serverError = "SERVER_ERROR"
    case validationError = "VALIDATION_ERROR"
    case notInitialized = "NOT_INITIALIZED"
    case alreadyVisible = "ALREADY_VISIBLE"
    case unknown = "UNKNOWN"
}

/// Result type for consent operations.
public enum ConsentResult<T> {
    case success(T)
    case error(message: String, code: ConsentErrorCode)
}

File: core/src/commonMain/kotlin/com/niceoneconsent/sdk/models/ConsentState.kt
package com.niceoneconsent.sdk.models

/**
 * Holds the current state of the consent popup flow.
 * This is an immutable data class; create copies via [copy] to update state.
 */
data class ConsentState(
    /** The user's entered age, or null if not yet entered. */
    val age: String = "",
    /** Set of selected purpose IDs. */
    val selectedPurposes: Set<String> = emptySet(),
    /** List of purposes fetched from the backend. */
    val purposes: List<Purpose> = emptyList(),
    /** Current language code. */
    val language: String = "en",
    /** Whether the SDK is currently loading data. */
    val isLoading: Boolean = false,
    /** Whether a consent submission is in progress. */
    val isSubmitting: Boolean = false,
    /** Current error message, or null if no error. */
    val error: String? = null,
    /** Whether the popup is currently visible. */
    val isPopupVisible: Boolean = false,
    /** Whether consent was successfully submitted. */
    val isConsentSubmitted: Boolean = false,
    /** Validation errors, keyed by field name. */
    val validationErrors: Map<String, String> = emptyMap()
)
